covasim/sim.py

```diff
@@ -381,25 +390,44 @@ def next(self, verbose=0):
-        self.results['n_exposed'][t]      = n_exposed
-        self.results['n_infectious'][t]   = n_infectious # Tracks total number infectious at this timestep
-        self.results['n_symptomatic'][t]  = n_symptomatic # Tracks total number symptomatic at this timestep
-        self.results['n_severe'][t]       = n_severe # Tracks total number of severe cases at this timestep
-        self.results['n_critical'][t]     = n_critical # Tracks total number of critical cases at this timestep
-        self.results['n_diagnosed'][t]    = n_diagnosed # Tracks total number of diagnosed cases at this timestep
-        self.results['bed_capacity'][t]   = n_severe/n_beds if n_beds>0 else None
+
+        self.results['n_susceptible'][t]  = n_susceptible * scale
+        self.results['n_exposed'][t]      = n_exposed * scale
+        self.results['n_infectious'][t]   = n_infectious * scale # Tracks total number infectious at this timestep
+        self.results['n_symptomatic'][t]  = n_symptomatic * scale # Tracks total number symptomatic at this timestep
+        self.results['n_severe'][t]       = n_severe * scale # Tracks total number of severe cases at this timestep
+        self.results['n_critical'][t]     = n_critical * scale # Tracks total number of critical cases at this timestep
+        self.results['n_diagnosed'][t]    = n_diagnosed * scale # Tracks total number of diagnosed cases at this timestep
+        self.results['bed_capacity'][t]   = (n_severe * scale)/n_beds if n_beds>0 else None
 
         # Update counts for this time step: flows
-        self.results['new_infections'][t]  = new_infections # New infections on this timestep
-        self.results['new_recoveries'][t]  = new_recoveries # Tracks new recoveries on this timestep
-        self.results['new_symptomatic'][t] = new_symptomatic
-        self.results['new_severe'][t]      = new_severe
-        self.results['new_critical'][t]    = new_critical
-        self.results['new_deaths'][t]      = new_deaths
+        self.results['new_infections'][t]  = new_infections * scale # New infections on this timestep
+        self.results['new_recoveries'][t]  = new_recoveries * scale # Tracks new recoveries on this timestep
+        self.results['new_symptomatic'][t] = new_symptomatic * scale
+        self.results['new_severe'][t]      = new_severe * scale
+        self.results['new_critical'][t]    = new_critical * scale
+        self.results['new_deaths'][t]      = new_deaths * scale
+
         self.t += 1
 
-        return
+    def resample(self, current_scale):
+        # Check if max pop was already reached
+        if current_scale * self["pop_size"] >= self.max_n:
+            return current_scale
+        susceptible = list(self.people.filter_in('susceptible'))
+        # Check if we've reached point when we want to resample and we didn't reach max population
+        if (len(susceptible) / len(self.people)) < self.resampe_threshold:
```

Comment:
Typo: resampe_threshold. Also, how about a `fraction()` method of `People` that does
```
ppl = list(self.people.filter_in(attr))
len(ppl)/len(self.people)
```
?
And is there a way we can avoid the slow operation of casting to list?